File: packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/hub/FlutterOutboxStatusEvent.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.amazonaws.amplify.amplify_datastore.types.hub

class FlutterOutboxStatusEvent(override val eventName: String, private val isEmpty: Boolean) : FlutterHubEvent {
    override fun toValueMap(): Map<String, Any> {
        return mapOf(
            "eventName" to this.eventName,
            "isEmpty" to this.isEmpty
        )
    }
}

File: packages/amplify_datastore/ios/Classes/types/hub/FlutterOutboxStatusEvent.swift
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

import Foundation
import Amplify
import AmplifyPlugins

struct FlutterOutboxStatusEvent: FlutterHubEvent {
    var eventName: String
    var isEmpty: Bool

    init(payload: HubPayload) throws {
        guard let outboxStatus = payload.data as? OutboxStatusEvent else {
                  throw FlutterDataStoreError.hubEventCast
              }
        self.eventName = shortEventName(eventName: payload.eventName)
        self.isEmpty = outboxStatus.isEmpty
    }

    func toValueMap() -> [String: Any] {
        return [
            "eventName": eventName,
            "isEmpty": isEmpty
        ]
    }
}
